chain-abci/src/enclave_bridge/mock.rs

```diff
@@ -234,7 +234,10 @@ impl EnclaveProxy for MockClient {
                 // FIXME
                 let plain_tx = PlainTxAux::decode(&mut txpayload.as_slice());
                 match (tx, plain_tx) {
-                    (_, Ok(PlainTxAux::TransferTx(maintx, witness))) => {
+                    (
+                        TxEnclaveAux::TransferTx { .. },
```

Comment:
how this fix crashes?

Reply:
The crash happen when wrapper tx type and payload tx type don't match, so after we fix the pattern match like this, that problematic case will go to the default branch, which will return an error rather than crash.